### PR TITLE
Add logsoftmax and fix for jax_funcify_join

### LINKED ISSUE #339: Add SoftmaxGrad Op implementation for JAX
@aseyboldt already has an implementation for this.

### PR DESCRIPTION
**Pull request by aseyboldt**
Add a LogSoftmax implementation for `jax_funcify`.

Also, this adds a jax tensor conversion for `jax_funcify_Join`. I'm not sure this is the best solution, it might also not be needed if optimizations are used. Without this change this fails:

```python
import pymc3 as pm
import aesara.tensor as aet
import aesara
import pymc3.sampling_jax
import numpy as np

with pm.Model() as model:
    a = pm.Normal('a', shape=(13, 3, 5))
    b = pm.Normal('b', shape=(17, 5, 7))
    pm.Normal('c', mu=aet.tensordot(a, b, axes=(2, 1)), shape=(13, 3, 17, 7))

with model:
    # fails with attribute access tuple.ndim
    trace = pymc3.sampling_jax.sample_numpyro_nuts()

# this doesn't fail, the join gets optimized away I think
func = model.logp_dlogp_function(mode=aesara.compile.mode.JAX)
func.set_extra_values({})
x = np.random.randn(func.size)

%%timeit
val, grad = func(x)
jax.device_get(grad)
```
cc @brandonwillard @OriolAbril 

closes: #340 and closes #339

### CODE REVIEW

**Review by OriolAbril — DISMISSED** [review]
Looks good, do you want me to rebase and continue with this forward?

**Review by brandonwillard — CHANGES_REQUESTED** [review]
We need to add a test before we can merge.

**Comment by twiecki** [comment]
@OriolAbril Can you take over and add a test?

**Comment by OriolAbril** [comment]
@brandonwillard should I add a test for Join? It looks like there is no specific test at all for Join in test jax. If so, which cases should it cover?

**Comment by brandonwillard** [comment]
> @brandonwillard should I add a test for Join? It looks like there is no specific test at all for Join in test jax. If so, which cases should it cover?

Absolutely, and especially if we don't already have a test for `Join`!

**Comment by OriolAbril** [comment]
I have added some tests using matrix types along both 0 and 1 axis but I don't really have any idea about what joins should be covered by the tests
